index.rst

```diff
@@ -39,4 +39,261 @@
+
+In August 2015 it was decided with `RFC-60 <https://jira.lsstcorp.org/browse/RFC-60>`_ that we should migrate the stack code (everything in ``lsst_distrib``) to support both Python 3 and Python 2.
+Packages outside of ``lsst_distrib`` have to support Python 3 but have the option of dropping Python 2 by filing an RFC demonstrating that they have no external users that will be affected.
+This document provides instructions on setting up a Python 3 environment and provides guidance on how to modify code to support both Python 2 and Python 3.
+
+A `preliminary guide on porting to Python 3 <http://dx.doi.org/10.5281/zenodo.60566>`_ was presented at the `LSST 2016 Project and Community Workshop <https://project.lsst.org/meetings/lsst2016/>`_.
+This document supercedes that content.
+
+Enabling Python 3
+=================
+
+At the time of writing Python 3 porters must use an `lsstsw <https://developer.lsst.io/build-ci/lsstsw.html>`_ environment to do the development.
+A new environment must be created and should not be shared with a Python 2 stack installation.
+In particular, do not share an EUPS database across both stacks.
+This is because the Python binary interfaces are not compatible between 2 and 3, and also EUPS currently has a bug resulting in database corruption when Python 2 attempts to read a database that was updated with Python 3.
+
+.. note ::
+    When sourcing the ``lsstsw`` ``setup.sh`` script, check that no EUPS environment variables
+    have been defined. EUPS will not overwrite these variables when doing the setup, leading to
+    possible confusion (and data corruption) over which EUPS database is active.
+
+Once the ``lsstsw`` git repository has been cloned, use ``bin/deploy -3`` to install a Python 3 ``miniconda``.
+You can then use ``lsstsw`` in the standard way.
+For example ``rebuild afw`` will build :lmod:`~lsst.afw` and all its dependencies, and ``rebuild -r tickets/DM-nnnn afw`` will build it using the specified ticket branch.
+
+Build System Caveats
+--------------------
+
+All LSST Data Management packages that follow the standard template and are integrated into the EUPS build system, use `SCons <http://scons.org>`_ for building.
+`SCons <http://scons.org>`_ does not yet work natively on Python 3.
+This means that ``SConstruct``, ``SConscript`` and :lmod:`sconsUtils` ``.cfg`` files should not assume that the Python running the SCons code is the same Python as that being used to run the stack code.
+This means that decisions should not be made by querying :mod:`distutils.sysconfig` or :data:`sys.executable` within the code.
+Instead, the ``python`` on ``$PATH`` must be queried instead, using :mod:`subprocess`.
+
+.. warning ::
+    At this time, to build a Python 3 stack, a ``python2.7`` binary must be available on the system and must be in the ``$PATH``.
+    ``python2.6`` is not suitable.
```

Comment:
I take this to mean that python 3.x *cannot* be used to install the stack. If that's correct then this is fine, but if python 3.x can be used instead of python 2.7 that should probably be explicitly stated.